### ISSUE TITLE
Target value used both as a feature and a target value in lin_model.fit(X, y)

### ISSUE THREAD

**Issue by PajLe** [MicrosoftDocs/sql-docs#7535]
On the [first code snippet](https://docs.microsoft.com/en-us/sql/machine-learning/tutorials/python-ski-rental-linear-regression-prepare-data?view=sql-server-ver15#explore-and-prepare-the-data), `columns` are set to include everything except `'Year'`.

Variable representing target column, `target` is equal to `'Rentalcount'`, which is also included in `columns`. This means that the `fit` call ([from the next step of the tutorial](https://docs.microsoft.com/en-us/sql/machine-learning/tutorials/python-ski-rental-linear-regression-train-model?view=sql-server-ver15#train-the-model)) will include target column in both X and y arguments:

```
# Fit the model to the training data.
lin_model.fit(train[columns], train[target])
```

Referencing to this [stats.stackexchange comment](https://stats.stackexchange.com/questions/573750/using-target-values-in-both-x-and-y-arguments-of-fitx-y?noredirect=1#comment1059040_573750), this shouldn't be correct. The variable `columns` shouldn't include the target column. 

My suggestion is to change `"Year"` to `"Rentalcount"` (or add `"Rentalcount"`) in 
```
columns = [c for c in columns if c not in ["Year"]]
```
at least as a quickfix, because it would lead to a bad model (around `0.2` r2 score). Still, the goal of the tutorial might not be to find the best model.

This change would also impact the next parts of the tutorial, like:
- [Part 3 - Make predictions](https://docs.microsoft.com/en-us/sql/machine-learning/tutorials/python-ski-rental-linear-regression-train-model?view=sql-server-ver15#make-predictions)
- [Part 4 - Create a stored procedure that generates the model](https://docs.microsoft.com/en-us/sql/machine-learning/tutorials/python-ski-rental-linear-regression-deploy-model?view=sql-server-ver15#create-a-stored-procedure-that-generates-the-model)
- [Part 4 - Create a stored procedure that makes predictions](https://docs.microsoft.com/en-us/sql/machine-learning/tutorials/python-ski-rental-linear-regression-deploy-model?view=sql-server-ver15#create-a-stored-procedure-that-makes-predictions)

PS: I'd be happy to do these changes if the suggestion is approved 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: e6126902-bcdc-3d8e-5977-ed58b8bf00f1
* Version Independent ID: 9dc69621-c9ce-840a-1dac-5da135f50515
* Content: [Python tutorial: Prepare data - SQL machine learning](https://docs.microsoft.com/en-us/sql/machine-learning/tutorials/python-ski-rental-linear-regression-prepare-data?view=sql-server-ver15)
* Content Source: [docs/machine-learning/tutorials/python-ski-rental-linear-regression-prepare-data.md](https://github.com/MicrosoftDocs/sql-docs/blob/live/docs/machine-learning/tutorials/python-ski-rental-linear-regression-prepare-data.md)
* Product: **sql**
* Technology: **machine-learning**
* GitHub Login: @WilliamDAssafMSFT
* Microsoft Alias: **wiassaf**

**Comment by WilliamAntonRohm** [comment]
@PajLe -- thank you for your thorough feedback.

@WilliamDAssafMSFT -- William, please look into this issue.

**Comment by WilliamDAssafMSFT** [comment]
@PajLe You are absolutely correct that this original ML sample is incorrectly including target columns in the training set. This is not a good example regardless of the model accuracy. We actually have a revamp of this example already in flight. I'll ping back when it is published and would be happy to have you review to make sure your concerns are addressed.
